pkgs/development/compilers/crystal/default.nix

```diff
@@ -238,10 +250,8 @@ rec {
   crystal_1_2 = generic {
     version = "1.2.2";
     sha256 = "sha256-nyOXhsutVBRdtJlJHe2dALl//BUXD1JeeQPgHU4SwiU=";
-    binary = crystal_1_1;
+    binary = if isAarch64Darwin then binaryCrystal_1_2 else crystal_1_1;
   };
 
   crystal = crystal_1_2;
-
-  crystal2nix = callPackage ./crystal2nix.nix { };
```

Comment:
Is this removal intentional? Seems unrelated

Reply:
It's intentional, since it's wrong and I assume it was just never removed when the file it's pointing to was. Do you want me to split it into a separate commit or PR?

Reply:
Separate commit would be great, thanks